tool/tctl/common/resource_command.go

```diff
@@ -404,6 +406,39 @@ func (rc *ResourceCommand) createUser(client auth.ClientI, raw services.UnknownR
 	return nil
 }
 
+// createAuthPreference implements `tctl create cap.yaml` command.
+func (rc *ResourceCommand) createAuthPreference(client auth.ClientI, raw services.UnknownResource) error {
```

Comment:
Actually, none of these checks are really meant to be guarantees. They are mere guidance which follows from the fact that the `origin` label can be manually assigned any value in the resource YAML (including `config-file` that'd override anything in place) so it doesn't make much sense to verify them on the auth service side, cf. [the RFD](https://github.com/gravitational/teleport/blob/master/rfd/0016-dynamic-configuration.md#implementation):
> Note that the `origin` label is not reserved or protected in any special way by the system and so it can be modified by the same means as other labels. User activity can disrupt the intended semantics and cause the label's value to cease to be representative of the actual origin or values of a resource.

Which is in turn based on [this discussion](https://github.com/gravitational/teleport/pull/5147#issuecomment-749715119):
> That's the trade-off I am OK to make. If user overwrites the labels set by the system, they may break the system. Kubernetes follows the similar approach by marking owner reference in the object controller creates. They set system labels and document their usage. If you update the system labels, the system may misbehave.

Reply:
I disagree with that choice, but let's go with the approved design.
Please leave a comment here with those links so that the next reader can understand the decision.